File: NBA_Server/src/main/java/entity/Match.java
package entity;

import java.util.ArrayList;
import java.util.List;

/**
 * 比赛实体类
 * 
 * created by JaneLDQ on 2015年5月17日 下午12:45:48
 */
public class Match {

	/**
	 * 比赛编号
	 */
	private String game_id;
	
	/**
	 * 赛季
	 */
	private String season;
	
	/**
	 * 日期
	 */
	private String date;
	
	/**
	 * 常规赛/季后赛
	 */
	private Boolean is_normal;
	
	/**
	 * 地点
	 */
	private String location;
	
	/**
	 * 主队
	 */
	private String home_team;
	
	/**
	 * 主队得分
	 */
	private Integer home_point;
	
	/**
	 * 客队
	 */
	private String guest_team;
	
	/**
	 * 客队得分
	 */
	private Integer guest_point;
	
	/**
	 * 比赛时长
	 */
	private String time="";
	
	/**
	 * 主队基本球员数据
	 */
	private List<MatchPlayerBasic> home_basic_list = new ArrayList<MatchPlayerBasic>();
	
	/**
	 * 主队高阶球员数据
	 */
	private List<MatchPlayerAdvanced> home_advanced_list = new ArrayList<MatchPlayerAdvanced>();
	
	/**
	 * 客队基本球员数据
	 */
	private List<MatchPlayerBasic> guest_basic_list = new ArrayList<MatchPlayerBasic>();
	
	/**
	 * 客队高阶球员数据
	 */
	private List<MatchPlayerAdvanced> guest_advanced_list = new ArrayList<MatchPlayerAdvanced>();
	
	/**
	 * 主队小节比分
	 */
	private List<Integer> home_pts = new ArrayList<Integer>();
	
	/**
	 * 客队小节比分
	 */
	private List<Integer> guest_pts = new ArrayList<Integer>();
	
	/**
	 * 无参构造函数
	 */
	public Match(){}

	public String getGame_id() {
		return game_id;
	}

	public String getSeason() {
		return season;
	}

	public void setSeason(String season) {
		this.season = season;
	}

	public void setGame_id(String game_id) {
		this.game_id = game_id;
	}

	public String getDate() {
		return date;
	}

	public void setDate(String date) {
		this.date = date;
	}

	public String getLocation() {
		return location;
	}

	public void setLocation(String location) {
		this.location = location;
	}

	public String getHome_team() {
		return home_team;
	}

	public void setHome_team(String home_team) {
		this.home_team = home_team;
	}

	public Integer getHome_point() {
		return home_point;
	}

	public void setHome_point(Integer home_point) {
		this.home_point = home_point;
	}

	public String getGuest_team() {
		return guest_team;
	}

	public void setGuest_team(String guest_team) {
		this.guest_team = guest_team;
	}

	public Integer getGuest_point() {
		return guest_point;
	}

	public void setGuest_point(Integer guest_point) {
		this.guest_point = guest_point;
	}

	public String getTime() {
		return time;
	}

	public void setTime(String time) {
		this.time = time;
	}

	public List<Integer> getHome_pts() {
		return home_pts;
	}

	public void setHome_pts(List<Integer> home_pts) {
		this.home_pts = home_pts;
	}

	public List<Integer> getGuest_pts() {
		return guest_pts;
	}

	public void setGuest_pts(List<Integer> guest_pts) {
		this.guest_pts = guest_pts;
	}

	public boolean isNormal() {
		return is_normal;
	}

	public void setNormal(boolean isNormal) {
		this.is_normal = isNormal;
	}

	public List<MatchPlayerBasic> getHome_basic_list() {
		return home_basic_list;
	}

	public void setHome_basic_list(List<MatchPlayerBasic> home_basic_list) {
		this.home_basic_list = home_basic_list;
	}

	public List<MatchPlayerAdvanced> getHome_advanced_list() {
		return home_advanced_list;
	}

	public void setHome_advanced_list(List<MatchPlayerAdvanced> home_advanced_list) {
		this.home_advanced_list = home_advanced_list;
	}

	public List<MatchPlayerBasic> getGuest_basic_list() {
		return guest_basic_list;
	}

	public void setGuest_basic_list(List<MatchPlayerBasic> guest_basic_list) {
		this.guest_basic_list = guest_basic_list;
	}

	public List<MatchPlayerAdvanced> getGuest_advanced_list() {
		return guest_advanced_list;
	}

	public void setGuest_advanced_list(List<MatchPlayerAdvanced> guest_advanced_list) {
		this.guest_advanced_list = guest_advanced_list;
	}
	
}

File: NBA_Scrape/model/match.py
# -*- coding: utf-8 -*-
# 比赛信息类(比赛编号、日期、比赛地点、主客队缩写、主客队得分、比赛时长)
class Match(object):

	# 限制Match的属性
	__slots__ = ('__game_id', '__date','__location','__home_team',
	'__home_point','__guest_team','__guest_point','__time'
	)

	def __init__(self):
		pass

	@property
	def game_id(self):
		return self.__game_id
	
	@game_id.setter
	def game_id(self, value):
		self.__game_id = value
		
	# 日期
	@property
	def date(self):
		return self.__date
	
	@date.setter
	def date(self,value):
		self.__date = value
	
	# 比赛地点
	@property
	def location(self):
		return self.__location
	
	@location.setter
	def location(self,value):
		self.__location = value
	
	# 主队缩写
	@property
	def home_team(self):
		return self.__home_team
		
	@home_team.setter
	def home_team(self, value):
		self.__home_team = value
		
	# 主队得分
	@property
	def home_point(self):
		return self.__home_point
	
	@home_point.setter
	def home_point(self, value):
		self._home_point = value
	
	# 客队缩写
	@property
	def guest_team(self):
		return self.__guest_team
	
	@guest_team.setter
	def guest_team(self, value):
		self.__guest_team = value
	
	# 客队得分
	@property
	def guest_point(self):
		return self._guest_point
		
	@guest_point.setter
	def guest_point(self, value):
		self.__guest_point = value
		
	@property
	def time(self):
		return self.__time
	
	@time.setter
	def time(self, value):
		self.__time = value
	
